### ISSUE TITLE
Plex metadata

### ISSUE THREAD

**Issue by deathbybandaid** [Sonarr/Sonarr#4960]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

N/A

### Describe the solution you'd like

Plex Beta release just dropped at time of writing. one of the items listed under "New" is 
````
NEW:
(Series Scanner) Allow the server owner to create “.plexmatch” files providing hints for more accurate matching of nonstandard file naming schemes (#13211)
````

As Sonarr/Radarr/Lidarr has metadata options for Kodi/Emby, it makes sense to also include the creation of this new `.plexmatch` file within series folders to help Plex more easily identify series.

https://forums.plex.tv/t/plex-media-server/30447/495

### Describe alternatives you've considered

N/A

### Anything else?

N/A

**Comment by brettpetch** [comment]
More info on the `.plexmatch` files, including formatting, dir structure, etc: https://support.plex.tv/articles/plexmatch/

**Comment by markus101** [comment]
Initially the plan is to support `.plexmatch` at the series level, which will include `title` and available IDs (`tvdbid`, `tmdbid`, and `imdbid`). If there is a need for additional information later we'll look at including that.

**Comment by brettpetch** [comment]
> I'll be happy with top-level series. There's a few shows that Plex likes to merge that over the years I've just manually fixed matches.

if you want a handy way to fix matches until it's built into the arr, you can use l3uddz's [Plexarr tool](https://github.com/l3uddz/plexarr)

**Comment by mynameisbogdan** [comment]
> Initially the plan is to support `.plexmatch` at the series level, which will include `title` and available IDs (`tvdbid`, `tmdbid`, and `imdbid`). If there is a need for additional information later we'll look at including that.

One weird issue I bumped into recently with the plex scanner is that it doesn't know how to parse multi-part mini-series, so I had to manually add something like this
```
ep: S01E01: Season 1/Example.1977.Special.Edition.Part1.BluRay.1080p.DTS.x264.NOGROUP.mkv
ep: S01E02: Season 1/Example.1977.Special.Edition.Part2.BluRay.1080p.DTS.x264.NOGROUP.mkv
ep: S01E03: Season 1/Example.1977.Special.Edition.Part3.BluRay.1080p.DTS.x264.NOGROUP.mkv
ep: S01E04: Season 1/Example.1977.Special.Edition.Part4.BluRay.1080p.DTS.x264.NOGROUP.mkv
```

I know it's outside of sonarr scope, but maybe adding support to define episodes and specials would be a great addition too.

One weird issue would be with tv shows that have two parts episodes that appear as one on tvdb.

**Comment by flick1999** [comment]
> Initially the plan is to support `.plexmatch` at the series level, which will include `title` and available IDs (`tvdbid`, `tmdbid`, and `imdbid`). If there is a need for additional information later we'll look at including that.

Any ETA on this, please? Thank you.

**Comment by flick1999** [comment]
Wow, dude, a thumbs down for a simple question, plus not even the courtesy of an answer? Or even a "go read the FAQ, idiot" response?

**Comment by markus101** [comment]
You asked the most common and least constructive thing here and now you’re upset you didn’t get a response, but a thumbs down for it.

No one should have to tell you to read a FAQ, the entire point of its existence is because they’re frequently asked questions that we’ve answered many times.

No one is going to provide you an ETA, as far as I know no one is actively working on this.

My previous comment was to explain how we want it implemented, not that I was going to implement it.

TLDR, don’t ask for an ETA, there is no ETA.

**Comment by thezak48** [comment]
> > Initially the plan is to support `.plexmatch` at the series level, which will include `title` and available IDs (`tvdbid`, `tmdbid`, and `imdbid`). If there is a need for additional information later we'll look at including that.
> 
> Any ETA on this, please? Thank you.

Simple temp fix for the mean time is put the tvdb id into the series folder if you are that desperate.

**Comment by brettpetch** [comment]
(Or use plexarr)

>

**Comment by flick1999** [comment]
Thank you, @brettpetch. I hadn't seen your earlier comment as someone marked it off-topic. I'll take a look at that.

And, @markus101, you're right. This is 100% on me. I've been spoiled by devs on other projects that, when they describe how a thing is to be done, it generally means they are working on said thing. And, to be clear, I did read your FAQs (https://wiki.servarr.com/sonarr/faq) and they don't address your release methodology. Also, your "report a bug" and "request a feature" links in the sidebar there are dead.

I also read your contributing page, your donation page, and several others before I made the ETA request, as literally nowhere could I locate your guidelines. So, yes, I asked a "common" question because it apparently isn't a "Frequently" Asked Question.  Your contribution page literally says "If you have any questions about any of this, please let us know." I had a question; I let you know.

What you have done is give me a good idea of the development team's attitude towards new folks here, which makes me less inclined to switch from my current tool and support this one.

Lastly, no, @thezak48, I'm not "desperate." I was merely trying to determine where a feature request was on the roadmap as part of my product evaluation. 

TLDR Thanks for the feedback. I'll show myself out.

**Comment by begna112** [comment]
> Initially the plan is to support `.plexmatch` at the series level, which will include `title` and available IDs (`tvdbid`, `tmdbid`, and `imdbid`). If there is a need for additional information later we'll look at including that.

I think it would be highly beneficial for episodes/seasons as well. After identifying files to episodes in sonarr, I'd like that information to be sync'd to Plex. 

For example, I had a show today that Plex mis-identified the last episode in the show as a previous episode and so I was thoroughly spoiled. If a plexmatch had been set up with each episode's file matched by sonarr, I would not have had this issue. 

I think it makes most sense to include a plexmatch at the series level as you described, but then also add a plexmatch at the season folder level for every show with file specifics.

**Comment by edrock200** [comment]
Not to muddy the waters, but at a series level a little known feature of the new TV and Movie agents is they can also scrape imdb/tvdb/tmdb id's from the folder names, so if you update your folder naming scheme accordingly, you can use it for proper plex matching today with Sonarr. This link shows the example for radarr, but same principle for Sonarr https://trash-guides.info/Radarr/Radarr-recommended-naming-scheme/

**Comment by ChaosBlades** [comment]
> @mynameisbogdan Because you're not following proper naming schemas. If you let Sonarr rename your media files, it'll be fine. Standardized naming for combined episodes even exists (e.g. S01E01-04 if you had all 4 parts combined in a single file). This is all already implemented in Sonarr, you're just not using it.

Not actually true as long as you don't want to put series ID in the title of the show. Blood+ being a good example. It names it Blood_ which results in some Korean drama and not the Anime even with the release year after the series name.. I have had several shows like this. Also I am discovering Plex always fails to find the show if it is Show, The instead of The Show. Having this file in the folder solves both those problems.

**Comment by KingIzzymon** [comment]
@ChaosBlades 

Idk if you've solved your issue yet, but I havent had a problem with Blood+ using this (thanks to [TRaSH)](https://trash-guides.info/Sonarr/Sonarr-recommended-naming-scheme/):
Settings > Media Management > Series Folder Format: ```{Series TitleYear} [tvdb-{TvdbId}]```

Results: ```Blood+ (2005) [tvdb-78960]```
